src/core/thread/mesh_forwarder.cpp

```diff
@@ -1250,7 +1256,11 @@ void MeshForwarder::HandleReceivedFrame(Mac::Frame &aFrame)
         if (payloadLength >= sizeof(Lowpan::MeshHeader) &&
             reinterpret_cast<Lowpan::MeshHeader *>(payload)->IsMeshHeader())
         {
+#if OPENTHREAD_FTD
             HandleMesh(payload, payloadLength, macSource, linkInfo);
+#else
+            assert(false);
```

Comment:
I don't think we should assert here. We don't control the sender implementation. We should gracefully drop packets that are not formatted properly.